mathesar/tests/api/test_record_api.py

```diff
@@ -1464,3 +1464,33 @@ def test_record_patch_unique_violation(create_patents_table, client):
         f'/api/db/v0/tables/{table.id}/constraints/{constraint_id}/'
     ).json()
     assert actual_constraint_details['name'] == 'NASA unique record PATCH_pkey'
+
+
+def test_record_patch_exclusion_violation(room_reservations_table, client):
+    table_name = 'ROOM exclusion record PATCH'
+    table = room_reservations_table(table_name)
+    id_column_id = table.get_column_name_id_bidirectional_map()['id']
+    columns_name_id_map = table.get_column_name_id_bidirectional_map()
+    # table.add_constraint(
+    #     ExclusionViolation(
+    #     what parameters should i pass ?
+    #     )
+    # )
```

Comment:
You could just use raw SQL to add the required constraint for now.

Reply:
I've attempted several SQL queries, but none  of them seems to be working successful. Additionally, I couldn't find any examples in the documentation on how to add exclusion constraints using SQL. Could you please provide guidance on adding exclusion constraints in SQL here @Anish9901

Reply:
You are facing this problem because the data type of the columns of the CSV are getting inferred as `TEXT` instead of `DATE`